src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/ChangeSignatureDialogInProcess.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls.Primitives;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Extensibility.Testing;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ChangeSignature;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using WindowsInput.Native;
+
+namespace Roslyn.VisualStudio.NewIntegrationTests.InProcess
+{
+    [TestService]
+    internal partial class ChangeSignatureDialogInProcess
+    {
+        private async Task<ChangeSignatureDialog?> TryGetDialogAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
+            return Application.Current.Windows.OfType<ChangeSignatureDialog>().SingleOrDefault();
+        }
+
+        private async Task ClickAsync(Func<ChangeSignatureDialog, ButtonBase> buttonAccessor, CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var dialog = await TryGetDialogAsync(cancellationToken);
+            AssertEx.NotNull(dialog);
+
+            Contract.ThrowIfFalse(await buttonAccessor(dialog).SimulateClickAsync(JoinableTaskFactory));
+        }
+
+        public async Task VerifyOpenAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                if (await TryGetDialogAsync(cancellationToken) is null)
```

Comment:
there's a `TryWaitForAsync` helper in one of the other services below - could be used here too.

Reply:
Maybe. I'm not thrilled with the current behavior but it matches the old code so left it for now.